include/aikido/control/TrajectoryExecutor.hpp

```diff
@@ -13,7 +12,13 @@ class TrajectoryExecutor
 public:
   virtual ~TrajectoryExecutor() = default;
 
-  /// Execute traj and set future upon completion.
+  /// Validate the traj in preparation for execution.
+  /// \param traj Trajectory to be validated
+  virtual void validate(trajectory::TrajectoryPtr traj) = 0;
```

Comment:
What exactly does it mean to validate a trajectory? Do we validate that it's collision free? Or that the robot's current state is in the same state as the beginning of the trajectory?

Reply:
Validating currently just means that the trajectory's DOFs match the DOFs from the skeleton that is being controlled by the `TrajectoryExecutor`. It could do more though -- I was thinking of it as a function for checking things that can be checked before executing the trajectory (like whether the DOFs match up). That said, I only added this to try and work around the deadlock issue.

I think verifying that the current state is in the same state as the beginning of the trajectory wouldn't fall under this category; we want to check that immediately before we actually execute the trajectory. I've added #272 for that.

I think validating whether the trajectory is collision-free shouldn't be the job of the executor, since it has no idea what the world looks like while it's executing.